docs/authorization.md

```diff
@@ -0,0 +1,149 @@
+
+  - `args`: derived from `@policy` directive arguments evaluation.
+  - `query`: the result of `query` policy property execution. The query is executed on the Stitch schema without effect of authorization policies (a.k.a. admin privileges). See examples below.
+
+- **_args_**: Arguments mapping. The key is argument name. The value is argument Graphql type. In this example the `userRole` argument can get value of `roles` claim from request JWT. (See `@policy` directive definition below).
+
+---
+
+#### Example 2: Policy with query
+
+```yaml
+metadata:
+  namespace: admin_ns
+  name: adminOnly
+type: opa
+code: |
+  default allow = false
+  allow {
+    input.query.user.roles[_] == "admin"
+  }
+args:
+  userId: ID
+query:
+  gql: |
+    query($id: ID!) {
+      user(id: $id) {
+        roles
+      }
+    }
+  variables:
+    id: '{args.userId}'
+```
+
+- **_query_**: Graphql query definition. It has 2 properties:
+
+  - `gql`: the [query](https://graphql.org/learn/queries) itself.
+  - `variables`: [variables](https://graphql.org/learn/queries/#variables) that are injected to the Graphql request. The values of variables can be either hardcoded or derived from `args`. In the last case the syntax is `"{args.userId}"` (See example above)
+
+---
+
+#### Example 3: Policy that depends on another policy
+
+Each policy is also available for evaluation as graphql query. It can be tested while debugging.
+If one policy is dependent on another the `query` property can include the following fragment:
+
+```graphql
+{
+  policy.another_policy_namespace___another_policy_name(<another_policy_args>) {
+    allow # As mentioned above the policy result is always object with the single boolean field "allow"
+  }
+}
+```
+
+The policy arguments should be set as the query arguments.
+
+In the example below the query has 2 parts: the first one fetches `roles` field from `user` and the second part evaluates `userIsActive`.
+
+```yaml
+metadata:
+  namespace: admin_ns
+  name: adminOnly
+type: opa
+code: |
+  default allow = false
+  allow {
+    input.args.roles[_] == "admin"
+    input.query.policy.another_ns___userIsActive.allow
+  }
+args:
+  userId: ID
+query:
+  gql: |
+    query($id: ID!) {
+      user(id: $id) {
+        roles
+      },
+      policy {
+        another_ns___userIsActive(userId: $id) {
+          allow
+        }
+      }
+    }
+  variables:
+    id: '{args.userId}'
+```
+
+---
+
+## Policy directive
+
+The policies are set using graphql directives. In order to attach policy to field the `@policy` directive should be set on the field.
+So if in type `User` there is field `phone` that should be accessible for user with admin role only the type definition should be as following:
+
+```graphql
+type User {
+  id: ID!
+  name: String!
+  phone: String @policy(namespace: "admin_ns", name: "admin_only", args: { roles: "{jwt.roles}" })
```

Comment:
Since you mentioned this is for the policy defined in example 1, we should probably stay consistent with the naming there:

```suggestion
  phone: String @policy(namespace: "billing", name: "adminOnly", args: { userRoles: "{jwt.roles}" })
```